src/registrar/views/utility/mixins.py

```diff
@@ -100,7 +100,14 @@ def can_access_other_user_domains(self, pk):
         if DomainInformation.objects.filter(id=pk).exists():
             requested_domain = DomainInformation.objects.get(id=pk)
 
-        if requested_domain.domain_application.status not in valid_domain_statuses:
+        # If no domain_application object exists and we are
+        # coming from the manage_domain dashboard, this is likely
+        # a transition domain.
+        domain_application = requested_domain.domain_application
```

Comment:
[Daves PR](https://github.com/cisagov/manage.get.gov/pull/1302) fixes this I believe, its a workaround until then.
This is an area I worked on a while ago...

Basically the manage domain (/admin) button does a check on the status, and these only exist on a domain_application object. While we do create those objects, they don't initially exist from the dashboard so it errors out (because it expects there to be one). 

The code above this checks if the user is authorized, if they clicked the button on the admin dashboard, and if they are navigating to the desired domain. Technically not accurate, but conceptually this is like adding "none" in the list of acceptable statuses